### ISSUE TITLE
Gradients not properly propagated when using generics

### ISSUE THREAD

**Issue by mabisson-code** [shader-slang/slang#9155]
# Issue Description

When a function takes a generic as a parameter, gradients are not properly propagated.

# Reproducer Code

This is a trimmed down version of the production code that triggered the issue.  It basically just copies the input tensor the output tensor.  This is what `apply_my_function_working()` is doing.

`apply_my_function_broken()` is doing the same thing, but going through a function that takes a generic parameter.  It's not even used, but this some how prevents the gradients from being propagated properly.

Slang code:

```
// repro.slang
interface IModifier {
    [Differentiable]
    float apply(float value);
};

void my_function<Modifier : IModifier>(
    no_diff uint tid,
    DiffTensorView input,
    DiffTensorView output,
    Modifier modifier
) {
    var value = input.load(tid);
    output.store(tid, value);
}


struct Modifier_Copy : IModifier, IDifferentiable {
    [Differentiable]
    [ForceInline]
    float apply(float value)
    {
        return value;
    }
};

[CUDAKernel]
[Differentiable]
[AutoPyBindCUDA]
void apply_my_function_broken(
    DiffTensorView input,
    DiffTensorView output,
) {
    const uint tid = cudaBlockIdx().x * cudaBlockDim().x + cudaThreadIdx().x;
    if (tid >= input.size(0)) {
        return;
    }

    var modifier = Modifier_Copy();
    my_function(tid, input, output, modifier);
}

[CUDAKernel]
[Differentiable]
[AutoPyBindCUDA]
void apply_my_function_working(
    DiffTensorView input,
    DiffTensorView output,
) {
    const uint tid = cudaBlockIdx().x * cudaBlockDim().x + cudaThreadIdx().x;
    if (tid >= input.size(0)) {
        return;
    }

    var value = input.load(tid);
    output.store(tid, value);
}
```

Python code:

```
import slangtorch
import torch

slang_module = slangtorch.loadModule("repro.slang")


for kernel_name in ['apply_my_function_broken', 'apply_my_function_working']:
    print(f"Testing {kernel_name}")
    kernel = getattr(slang_module, kernel_name)

    input = torch.randn(2, device='cuda', dtype=torch.float32)
    output = torch.full_like(input, -1.0)

    # Forward pass.
    print(f"    input: {input}")
    print(f"    output before: {output}")
    kernel(
        input=input,
        output=output,
        ).launchRaw(blockSize=(32, 1, 1), gridSize=(1, 1, 1))
    print(f"    output after: {output}")
    print("    --------------------------------")

    # Backward pass.
    output_grad = torch.randn_like(output)
    input_grad = torch.full_like(input, -1.0)

    print(f"    output_grad: {output_grad}")
    print(f"    input_grad before: {input_grad}")
    kernel.bwd(
        input=(input, input_grad),
        output=(output, output_grad),
        ).launchRaw(blockSize=(32, 1, 1), gridSize=(1, 1, 1))
    print(f"    input_grad after: {input_grad}")
    print("    --------------------------------")
```

# Expected Behavior

Since this is just a copy, we would expect the gradients of input to be the same as the gradients of output after running the backward kernel.

# Actual Behavior

No gradients are computed / propagated.  The generated backward kernel is actually empty:

```
__global__ void __kernel__apply_my_function_broken_bwd_diff(DiffTensorView_0 input_1, DiffTensorView_0 output_1)
{
    return;
}
```

# Environment
 - Slang Version: 2025.21.2
 - OS: Windows

**Comment by mabisson-code** [comment]
This is a bug in the repro code: `my_function` should be marked `[Differentiable]`.  When it is, this code no longer shows any issue.

I tried reproing the bug using something closer to the original code, and I was able to fix the original code.  The main thing that was missing was marking the interface as differentiable.

So in the end, there is no bug to fix in Slang.  Closing this ticket.
